simulation/traffic_simulator/src/api/api.cpp

```diff
@@ -397,4 +423,40 @@ void API::requestLaneChange(
   entity_manager_ptr_->requestLaneChange(name, target, trajectory_shape, constraint);
 }
 
+void API::requestSpeedChange(const std::string & name, double target_speed, bool continuous)
+{
+  assert(entity_manager_ptr_);
```

Comment:
`assert` is simple and convenient, but most of the time, scenario_simulator_v2 is built with release build setting that disable asserts , so it's practically useless.
It is recommended to throw an exception, instead of using `assert`.

Reply:
I've changed it according to your feedback -> a0e78a1744acd16f664c71bfc9006bdadd22baee